app-modules/service-management/src/Filament/Resources/ServiceRequestResource/Pages/ListServiceRequests.php

```diff
@@ -164,11 +165,33 @@ public function table(Table $table): Table
                         )
                     ))
                     ->preload(),
-                Filter::make('Unassigned')
+                SelectFilter::make('assignedTo')
+                    ->searchable()
+                    ->preload()
+                    ->options(fn () => [
+                        '' => ['unassigned' => 'Unassigned'],
+                        'Assigned to' => User::query()->limit(50)->pluck('name', 'id')->toArray(), // Initially load 50 users
+                    ])
+                    ->getSearchResultsUsing(fn (string $search): array => ['Assigned To' => User::query()->where(new Expression('lower(name)'), 'like', '%' . strtolower($search) . '%')->take(50)->pluck('name', 'id')->toArray()])
                     ->query(
-                        fn (Builder $query) => $query->whereDoesntHave('assignedTo', function (Builder $query) {
-                            $query->where('status', ServiceRequestAssignmentStatus::Active);
-                        })
+                        fn (Builder $query, array $data) => $query
+                            ->when(
+                                empty($data['value']),
+                                fn (Builder $baseQuery) => $baseQuery
+                            )
+                            ->when(
+                                $data['value'] === 'unassigned',
+                                fn (Builder $baseQuery) => $baseQuery->whereDoesntHave('assignedTo', function (Builder $query) {
+                                    $query->where('status', ServiceRequestAssignmentStatus::Active);
+                                })
+                            )
+                            ->when(
+                                ! empty($data['value']) && $data['value'] !== 'unassigned',
+                                fn (Builder $baseQuery) => $baseQuery->whereHas(
+                                    'assignedTo.user',
+                                    fn (Builder $userQuery) => $userQuery->where('id', $data['value'])
+                                )
```

Comment:
I think we similarly want to check that the assignment is active. Similiar to what you did above with `->where('status', ServiceRequestAssignmentStatus::Active)`